pkg/generator/generate.go

```diff
@@ -596,13 +597,23 @@ func (g *schemaGenerator) generateStructType(
 				"skipping validation code for them since we don't know their types")
 		}
 		valueType := codegen.Type(codegen.EmptyInterfaceType{})
-		var err error
-		_, ok := (*t.AdditionalProperties).(bool)
-		if !ok && t.AdditionalProperties != nil {
-			casted := (*t.AdditionalProperties).(*schemas.Type)
-			valueType, err = g.generateType(casted, nil); 
-			if err != nil {
-				return nil, err
+
+		if t.AdditionalProperties != nil {
+			_, ok := (*t.AdditionalProperties).(bool)
+			if !ok && t.AdditionalProperties != nil {
```

Comment:
are you sure you need double check that AdditionalProperties != nil ?
```suggestion
			if !ok {
```

Reply:
also I would suggest using guard if statements instead of nesting